doc/adr/0006-selecting-facilitators.md

```diff
@@ -0,0 +1,21 @@
+# 6. Selecting facilitators
+
+Date: 2022-01-17
+
+## Status
+
+Proposed
+
+## Context
+
+Facilitators of L1 should be selected by L0 layer.
+
+## Decision
+
+L0 nodes can select next facilitators by looking at the active peer list from L1.
+These facilitators are put to Global Snapshot so L1 can fetch it and run the consensus.
+L0 nodes will validate signatures of previously selected facilitators when blocks are received from l1.
```

Comment:
Guys there is one possible edge case @buckysballs, @marcinwadon, @TheMMaciek, @tbekas 

1. We have selected facilitators `a`, `b`,`c`.
2. L0 receives blocks with tips and at some point is ready to create a snapshot.
3. L0 creates a snapshot and selects the next facilitators:  `d`, `e`, `f`.
4. After selecting the next facilitators and creating a snapshot L0 receives a delayed block from facilitator `b` which is obviously invalid according to that logic because that guy is not a facilitator anymore.

Reply:
Assuming these L1 facilitators? Regardless of whether an L1 facilitator is active or not, if a block doesn't make it into any snapshot proposal and misses the appropriate height window then the block effectively missed its chance and the data (transactions etc.) need to be re-enqueued into the mempool

Reply:
Ok, I will post it inside ADR